File: php/shapes/Rectangle.php
<?php

//require_once "Shape.php";
//require_once "Quadrant.php";

class Rectangle extends Shape
{
    private float $width;
    private float $height;

    public function __construct(Quadrant $quadrant, float $width, float $height)
    {
        parent::__construct($quadrant);
        $this->width = $width;
        $this->height = $height;
    }

    function isInShape(float $x, float $y): bool
    {
        return $this->isInQuadrant($x, $y) &&
            $this->quadrant->getX() * $x <= $this->width && $this->quadrant->getY() * $y <= $this->height;
    }
}

File: src/main/java/com/moleus/web/service/areaCheck/shapes/Rectangle.java
package com.moleus.web.service.areaCheck.shapes;

import com.moleus.web.service.areaCheck.quadrant.Point;
import com.moleus.web.service.areaCheck.quadrant.Quadrant;

public class Rectangle extends AbstractShape {
    private final float width;
    private final float height;

    public Rectangle(Quadrant quadrant, float width, float height) {
        super(quadrant);
        this.width = width;
        this.height = height;
    }

    @Override
    public boolean isInShape(Point point) {
        return super.isInQuadrant(point) &&
            this.quadrant.x * point.x() <= this.width && this.quadrant.y * point.y() <= this.height;
    }
}
